src/wasm32_bindgen.rs

```diff
@@ -64,9 +64,15 @@ fn getrandom_init() -> Result<RngSource, Error> {
         // `crypto.getRandomValues`, but if `crypto` isn't defined we assume
         // we're in an older web browser and the OS RNG isn't available.
 
-        let crypto = self_.crypto();
+        let mut crypto = self_.crypto();
         if crypto.is_undefined() {
-            return Err(BINDGEN_CRYPTO_UNDEF);
+            // If crypto is undefined, we can also check if msCrypto is defined, 
+            // in case it's Internet Explorer
+            crypto = self_.ms_crypto();
+            
+            if crypto.is_undefined() {
+                return Err(BINDGEN_CRYPTO_UNDEF);
+            }
```

Comment:
We can make this logic more straightforward using Rust's match guards:
```rust
        let crypto: BrowserCrypto = match (self_.crypto(), self_.ms_crypto()) {
            (crypto, _) if !crypto.is_undefined() => crypto.into(),
            (_, crypto) if !crypto.is_undefined() => crypto.into(),
            _ => return Err(BINDGEN_CRYPTO_UNDEF),
        };

        // Test if `crypto.getRandomValues` is undefined as well
        if crypto.get_random_values_fn().is_undefined() {
            return Err(BINDGEN_GRV_UNDEF);
        }
```

Reply:
I really like that!